## Problem Statement :

Given is a list of business descriptions for around 24,000 companies (provided in the file company descriptions). There are 2 types of descriptions – long and short. Use your discretion while using either. Perform the following task using the file:  
Classify these companies based on their business descriptions to only one of the industries from
the industry labels given (provided in the file: Industry Segments – Top 10 Keywords).   

## Sequential steps followed during problem solving :
1. Exploring the dataset.
2. Variables Identification.
3. Missing value Treatement.
4. Text Preprocessing.
5. Applying algorithm (first on short description column).
6. Follow step 3, 4, 5 for company description (long) column.
7. Finally compare result on basis of both columns.


## 1. Exploring the datasets :


In [1]:
# importing common used library

import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
# importing datasets on google colab

#from google.colab import files
#files.upload()

In [3]:
#files.upload()

In [4]:
# reading the datasets

keyword = pd.read_excel("Industry Segments - Top 10 Keywords.xlsx")
company = pd.read_excel("company descriptions.xlsx")

# copying the datasets so that after any manupulation we still have original dataset

keyword_copy=keyword.copy()
company_copy=company.copy()

# checking dimension of datasets

print("keyword dataset dimension : ",keyword.shape)
print("company dataset dimension : ",company.shape)

keyword dataset dimension :  (30, 2)
company dataset dimension :  (19965, 3)


In [5]:
# accessing first five rows of keyword dataset to see how it looks like

print("*"*71)
print("first five rows of keyword data :-")
print("*"*71)
keyword.head()

***********************************************************************
first five rows of keyword data :-
***********************************************************************


Industry segment  \
0          Aerospace and defense   
1       Agriculture and forestry   
2             Biopharmaceuticals   
3      Business support services   
4  Communications and networking   

                                     Top 10 keywords  
0  Security, systems, video, surveillance, servic...  
1  Service, cleantech, water, agriculture, traits...  
2  Developer, treatment, drug, diseases, technolo...  
3  Service, platform, online, management, data, m...  
4  Service, wireless, network, data, internet, ap...

In [6]:
# accessing first five rows of company dataset to see how it looks like

print("*"*71)
print("first five rows of company data:-")
print("*"*71)
company.head()

***********************************************************************
first five rows of company data:-
***********************************************************************


company_name                          company_short_description  \
0        Codementor  Codementor is an online marketplace connecting...   
1           AgShift  AgShift is designing world's most advanced aut...   
2            Shipsi  Shipsi empowers any retailer with the ability ...   
3          OpenNews  OpenNews helps a global network of developers,...   
4  Biobot Analytics  Biobot Analytics analyzes city sewage to estim...   

                                 company_description  
0  Codementor provides live 1:1 help for software...  
1  AgShift solution blends Deep Learning with Com...  
2  Shipsi empowers any retailer with the ability ...  
3  We're helping a global network of developers, ...  
4  Biobot Analytics analyzes sewage to estimate o...

## 2. Variables Identification:



In [7]:
# checking datatype of each columns and any missing values

print(keyword.info())
print("*"*71)
print("*"*71)
print(company.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 2 columns):
Industry segment    28 non-null object
Top 10 keywords     27 non-null object
dtypes: object(2)
memory usage: 608.0+ bytes
None
***********************************************************************
***********************************************************************
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19965 entries, 0 to 19964
Data columns (total 3 columns):
company_name                 19965 non-null object
company_short_description    19965 non-null object
company_description          19237 non-null object
dtypes: object(3)
memory usage: 468.1+ KB
None


## Observations :
1. Keyword datset consist of two columns, Idustry segment column has 28 non-missing values/rows out of 30 while Top 10 keywords column has 27 non-missing values/rows out of 30.
2. Keyword datset consist of three columns, company_name column has 19965 non-missing values/rows out of 19965 while company_short_description column has 19965 non-missing values/rows out of 19965 and company_description column has 19237 non-missing values/rows out of 19965.
3. The datatype of all columns in both datasets are Object.


## 3. Missing Value Treatment:

In [8]:
# Check the percentage of null values per variable

print("% of null values for each column:\n",keyword.isnull().sum()/keyword.shape[0]*100)
print("*"*71)
print("*"*71)
print("% of null values for each column:\n",company.isnull().sum()/company.shape[0]*100)

% of null values for each column:
 Industry segment     6.666667
Top 10 keywords     10.000000
dtype: float64
***********************************************************************
***********************************************************************
% of null values for each column:
 company_name                 0.000000
company_short_description    0.000000
company_description          3.646381
dtype: float64


## Observations :

1. Dropping all rows containing NAN from keyword dataset because we can't fill directly any value (like mode etc) because keywords is related to that particular company profile.
2. Not dropping any values from company dataset because I am considering only two columns (company_name, company_short_description) and there is no missing values for these two columns and further we can extend to third column.


In [9]:
# dropping missing values from keyword dataset as we can't fill directly value (like mode etc) because keyword is related to that particular company profile

keyword = keyword.dropna()
print("keyword dataset new dimension : ",keyword.shape)

# considering only two columns

company_short = company[['company_name', 'company_short_description']]
print("company_short dataset dimension : ",company_short.shape)

keyword dataset new dimension :  (27, 2)
company_short dataset dimension :  (19965, 2)


## 4. Text Preprocessing : 

In [10]:
import re
import string

# removing number, punctuation etc
# converting into lower case

alphanumeric = lambda x: re.sub(r"""\w*\d\w*""", ' ', x)
punc_lower = lambda x: re.sub('[%s]' % re.escape(string.punctuation), ' ', x.lower())

keyword['Top 10 keywords'] = keyword['Top 10 keywords'].map(punc_lower)

company_short['company_short_description'] = company_short.company_short_description.map(alphanumeric).map(punc_lower)
company_short.head(3)

C:\Users\Purushottam\Anaconda3\lib\site-packages\ipykernel_launcher.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if sys.path[0] == '':


company_name                          company_short_description
0   Codementor  codementor is an online marketplace connecting...
1      AgShift  agshift is designing world s most advanced aut...
2       Shipsi  shipsi empowers any retailer with the ability ...

In [11]:
keyword.head(3)

Industry segment                                    Top 10 keywords
0     Aerospace and defense  security  systems  video  surveillance  servic...
1  Agriculture and forestry  service  cleantech  water  agriculture  traits...
2        Biopharmaceuticals  developer  treatment  drug  diseases  technolo...

In [12]:
# importing library for text preprocessing part

import nltk

# this 2nd line doesn't require in jupyter notebook after downloading once
# but in google golab it requires again if we end/close this notebook

nltk.download('stopwords')

[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\Purushottam\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


True

In [13]:
# importing stopwords 
# the reason for removing stopwords is that if some company description have large number of stopwords then probability will decrease due to this

# even this step can be performed using CountVectorizer (using stop word parameter, I had done in clustering part) 
# and there is no need of CountVectorizer here

from nltk.corpus import stopwords

stop = stopwords.words('english')

In [14]:
# removing stopwords

company_short['company_short_description'] = company_short['company_short_description'].apply(lambda x: ' '.join([word for word in x.split() if word not in (stop)]))
company_short.head(3)

C:\Users\Purushottam\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


company_name                          company_short_description
0   Codementor  codementor online marketplace connecting devel...
1      AgShift  agshift designing world advanced autonomous fo...
2       Shipsi  shipsi empowers retailer ability offer easy “i...

In [15]:
# for tokenization, We have to download punkt

nltk.download('punkt')

[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\Purushottam\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!


True

In [16]:
# tokenizing short description of each word

from nltk.tokenize import word_tokenize

company_short['tokenized_word'] = company_short['company_short_description'].apply(word_tokenize)
company_short.head(3)

company_name                          company_short_description  \
0   Codementor  codementor online marketplace connecting devel...   
1      AgShift  agshift designing world advanced autonomous fo...   
2       Shipsi  shipsi empowers retailer ability offer easy “i...   

                                      tokenized_word  
0  [codementor, online, marketplace, connecting, ...  
1  [agshift, designing, world, advanced, autonomo...  
2  [shipsi, empowers, retailer, ability, offer, e...

In [17]:
keyword['tokenized'] = keyword['Top 10 keywords'].apply(word_tokenize)
keyword.head(3)

Industry segment  \
0     Aerospace and defense   
1  Agriculture and forestry   
2        Biopharmaceuticals   

                                     Top 10 keywords  \
0  security  systems  video  surveillance  servic...   
1  service  cleantech  water  agriculture  traits...   
2  developer  treatment  drug  diseases  technolo...   

                                           tokenized  
0  [security, systems, video, surveillance, servi...  
1  [service, cleantech, water, agriculture, trait...  
2  [developer, treatment, drug, diseases, technol...

## 5. Applying Algorithm :

1. Taking first row from company dataset.
2. Taking first row from keyword dataset.
3. Now finding number of matching keyword (tokenized column) with respect to tokenized_word column (in company_short table).
4. Dividing number obtained in 5.3 by number of words in that particular row of tokenized_word column (in company_short table).
5. Running 5.2, for all rows of kwyword dataset.
6. Finding max probability after running 5.5.
7. Running 5.1, for all rows of company dataset.

Note :
1. There may be exception as for some iteration, probability value zero and this will happen when there is no matching words.

Another Algorithm :
1. Tf-idf can also be used to find probability etc (This concept, I have used in clustering).

In [18]:
# actually this step I have added after running just next line of code and next line is giving zero division error
# it means that some(in this dataset there is only one such row whose index is 8671) of the values in company_short_description are empty string ('') 

print(company_short.loc[8670:8672,:])
company_short=company_short[company_short.company_short_description != '']

      company_name                          company_short_description  \
8670     Culture22  award winning creative interactive technology ...   
8671          C3RS                                                      
8672  ITA Software  ita software airline services provider helping...   

                                         tokenized_word  
8670  [award, winning, creative, interactive, techno...  
8671                                                 []  
8672  [ita, software, airline, services, provider, h...  


In [19]:
# for storing maximum probability value
result = []

# for storing possible Industry segment for each of the company
result1 = []

# for each observation of company dataset, I am going to run all observations of keyword dataset
for j in company_short.loc[:,'tokenized_word']:
  # for storing probability value for each rows of keyword dataset 
  li = []
  # initializing sum equals to zero, it will keep track of counting of matcing words
  sum = 0
  
  for i in keyword.iloc[:,2]:
    # running loop for each keyword of each row of keyword dataset
    for k in i:
      # counting occurence of each particular word in tokenized_word column of company_short dataset
      count = j.count(k)
      sum += count

    # finding probability
    sum = sum/len(j)
    li.append(sum)
    
  #print(li)
  #print(len(li))
  #print(li.index(max(li)))
  #print(max(li))

  # now taking out maximum probability value from li list and appending to result list
  result.append(max(li))
  # now taking out index corresponding to maximum probability value and then Industry segment corresponding to this and appending to result1 list
  result1.append(keyword.loc[li.index(max(li)),'Industry segment'])


In [20]:
# adding two new columns in company_short dataset

company_short['probab'] = result
company_short['tag/Industry_segment'] = result1

In [21]:
# finally seeing tag/Industry_segment corresponding to each company
# we can observe 5th row, probab value and it is zero meaning that there is no matching keyword and hence there is no tag
# for such case, I am replacing with 'Others'

company_short.head()

company_name                          company_short_description  \
0        Codementor  codementor online marketplace connecting devel...   
1           AgShift  agshift designing world advanced autonomous fo...   
2            Shipsi  shipsi empowers retailer ability offer easy “i...   
3          OpenNews  opennews helps global network developers journ...   
4  Biobot Analytics  biobot analytics analyzes city sewage estimate...   

                                      tokenized_word    probab  \
0  [codementor, online, marketplace, connecting, ...  0.142857   
1  [agshift, designing, world, advanced, autonomo...  0.099242   
2  [shipsi, empowers, retailer, ability, offer, e...  0.000000   
3  [opennews, helps, global, network, developers,...  0.076923   
4  [biobot, analytics, analyzes, city, sewage, es...  0.000000   

            tag/Industry_segment  
0              Media and content  
1  Medical devices and equipment  
2          Aerospace and defense  
3  Communications and networking  
4          Aerospace and defense

In [22]:
# finding all such where probab is zero and then assigning Others to tag/Industry_segment column

company_short.loc[company_short['probab'] == 0.000000, 'tag/Industry_segment'] = 'Others'

In [23]:
# now we can see 5th row, column tag/Industry_segment column

company_short.head()

company_name                          company_short_description  \
0        Codementor  codementor online marketplace connecting devel...   
1           AgShift  agshift designing world advanced autonomous fo...   
2            Shipsi  shipsi empowers retailer ability offer easy “i...   
3          OpenNews  opennews helps global network developers journ...   
4  Biobot Analytics  biobot analytics analyzes city sewage estimate...   

                                      tokenized_word    probab  \
0  [codementor, online, marketplace, connecting, ...  0.142857   
1  [agshift, designing, world, advanced, autonomo...  0.099242   
2  [shipsi, empowers, retailer, ability, offer, e...  0.000000   
3  [opennews, helps, global, network, developers,...  0.076923   
4  [biobot, analytics, analyzes, city, sewage, es...  0.000000   

            tag/Industry_segment  
0              Media and content  
1  Medical devices and equipment  
2                         Others  
3  Communications and networking  
4                         Others

## 6. Applying same techinque to column company_description (long description) :

In [24]:
# selecting two columns only

# under section 3, we have already observed missing values
# we again can't fill any value (like mode etc) so I have decided to drop such rows

# another solution, we can fill short description value into long description value

company_long = company[['company_name', 'company_description']]
company_long = company_long.dropna()
print("company_long dataset dimension : ",company_long.shape)

company_long dataset dimension :  (19237, 2)


In [25]:
import re
import string

# removing number, punctuation etc
# converting into lower case

alphanumeric = lambda x: re.sub(r"""\w*\d\w*""", ' ', x)
punc_lower = lambda x: re.sub('[%s]' % re.escape(string.punctuation), ' ', x.lower())

company_long['company_description'] = company_long.company_description.map(alphanumeric).map(punc_lower)
company_long.head(3)

company_name                                company_description
0   Codementor  codementor provides live     help for software...
1      AgShift  agshift solution blends deep learning with com...
2       Shipsi  shipsi empowers any retailer with the ability ...

In [26]:
# removing stopwords

company_long['company_description'] = company_long['company_description'].apply(lambda x: ' '.join([word for word in x.split() if word not in (stop)]))
company_long.head(3)

company_name                                company_description
0   Codementor  codementor provides live help software develop...
1      AgShift  agshift solution blends deep learning computer...
2       Shipsi  shipsi empowers retailer ability offer easy “i...

In [27]:
# tokenizing company description of each word

company_long['tokenized_word_1'] = company_long['company_description'].apply(word_tokenize)
company_long.head(3)

company_name                                company_description  \
0   Codementor  codementor provides live help software develop...   
1      AgShift  agshift solution blends deep learning computer...   
2       Shipsi  shipsi empowers retailer ability offer easy “i...   

                                    tokenized_word_1  
0  [codementor, provides, live, help, software, d...  
1  [agshift, solution, blends, deep, learning, co...  
2  [shipsi, empowers, retailer, ability, offer, e...

In [28]:
# removing rows if it contains empty string

company_long=company_long[company_long.company_description != '']

In [29]:
# applying same algorithm 

result = []
result1 = []

for j in company_long.loc[:,'tokenized_word_1']:
  
  li = []
  
  sum = 0
  
  for i in keyword.iloc[:,2]:
    for k in i:
      
      count = j.count(k)
      sum += count

    
    sum = sum/len(j)
    li.append(sum)
    
  
  #print(li)
  #print(len(li))
  #print(li.index(max(li)))
  #print(max(li))

  result.append(max(li))
  result1.append(keyword.loc[li.index(max(li)),'Industry segment'])


In [30]:
company_long['probab_1'] = result
company_long['tag_1'] = result1

In [31]:
company_long.loc[company_long['probab_1'] == 0.000000, 'tag_1'] = 'Others'

In [32]:
company_long.head(5)

company_name                                company_description  \
0        Codementor  codementor provides live help software develop...   
1           AgShift  agshift solution blends deep learning computer...   
2            Shipsi  shipsi empowers retailer ability offer easy “i...   
3          OpenNews  helping global network developers journalists ...   
4  Biobot Analytics  biobot analytics analyzes sewage estimate opio...   

                                    tokenized_word_1  probab_1  \
0  [codementor, provides, live, help, software, d...  0.035720   
1  [agshift, solution, blends, deep, learning, co...  0.044779   
2  [shipsi, empowers, retailer, ability, offer, e...  0.032292   
3  [helping, global, network, developers, journal...  0.074074   
4  [biobot, analytics, analyzes, sewage, estimate...  0.073395   

                                       tag_1  
0                         Biopharmaceuticals  
1                                   Software  
2               Wholesale trade and shipping  
3                          Media and content  
4  Medical software and information services

## 7. Observing and comparing which columns out of two works well:

In [33]:
company_short.head(20)

company_name  \
0                Codementor   
1                   AgShift   
2                    Shipsi   
3                  OpenNews   
4          Biobot Analytics   
5                  NetClerk   
6                 Satellier   
7                  MedCases   
8                 SpeedDate   
9     Raven Biotechnologies   
10             Kids on 45th   
11          Xtreme Spectrum   
12     WindMIL Therapeutics   
13                   Alacra   
14                  PetDesk   
15               BrightDime   
16  Comstellar Technologies   
17                    Lot18   
18                     ROAM   
19                 Homebase   

                            company_short_description  \
0   codementor online marketplace connecting devel...   
1   agshift designing world advanced autonomous fo...   
2   shipsi empowers retailer ability offer easy “i...   
3   opennews helps global network developers journ...   
4   biobot analytics analyzes city sewage estimate...   
5   netclerk application service provider asp requ...   
6   satellier provides workshare solutions archite...   
7   medcases internet based provider interactive m...   
8   speeddate com online webcam enhanced dating si...   
9   raven biotechnologies focused development mono...   
10  building box kid everyday essentials low price...   
11  xtreme spectrum provider ultra wideband uwb so...   
12  windmil therapeutics developing cell therapies...   
13                     information workflow solutions   
14  petdesk offers crm software pet care providers...   
15  brightdime offers everything company needs suc...   
16  comstellar technologies inc building major tec...   
17  online marketer seller wine gourmet goods base...   
18   roam global community next generation adventurer   
19          homebase easiest way track hours schedule   

                                       tokenized_word    probab  \
0   [codementor, online, marketplace, connecting, ...  0.142857   
1   [agshift, designing, world, advanced, autonomo...  0.099242   
2   [shipsi, empowers, retailer, ability, offer, e...  0.000000   
3   [opennews, helps, global, network, developers,...  0.076923   
4   [biobot, analytics, analyzes, city, sewage, es...  0.000000   
5   [netclerk, application, service, provider, asp...  0.124998   
6   [satellier, provides, workshare, solutions, ar...  0.000000   
7   [medcases, internet, based, provider, interact...  0.140625   
8   [speeddate, com, online, webcam, enhanced, dat...  0.200593   
9   [raven, biotechnologies, focused, development,...  0.444444   
10  [building, box, kid, everyday, essentials, low...  0.000000   
11  [xtreme, spectrum, provider, ultra, wideband, ...  0.000000   
12  [windmil, therapeutics, developing, cell, ther...  0.142857   
13                 [information, workflow, solutions]  0.333333   
14  [petdesk, offers, crm, software, pet, care, pr...  0.300000   
15  [brightdime, offers, everything, company, need...  0.111111   
16  [comstellar, technologies, inc, building, majo...  0.099994   
17  [online, marketer, seller, wine, gourmet, good...  0.101001   
18  [roam, global, community, next, generation, ad...  0.166667   
19   [homebase, easiest, way, track, hours, schedule]  0.000000   

                         tag/Industry_segment  
0                           Media and content  
1               Medical devices and equipment  
2                                      Others  
3               Communications and networking  
4                                      Others  
5                           Media and content  
6                                      Others  
7   Medical software and information services  
8                                   Retailers  
9                          Biopharmaceuticals  
10                                     Others  
11                                     Others  
12                         Biopharmaceuticals  
13  Medical software and information services  
14  Medical software and information 

In [34]:
company_long.head(20)

company_name  \
0                Codementor   
1                   AgShift   
2                    Shipsi   
3                  OpenNews   
4          Biobot Analytics   
5                  NetClerk   
6                 Satellier   
7                  MedCases   
8                 SpeedDate   
9     Raven Biotechnologies   
11          Xtreme Spectrum   
12     WindMIL Therapeutics   
13                   Alacra   
14                  PetDesk   
15               BrightDime   
16  Comstellar Technologies   
17                    Lot18   
19                 Homebase   
20                     Miro   
21                 Tenebril   

                                  company_description  \
0   codementor provides live help software develop...   
1   agshift solution blends deep learning computer...   
2   shipsi empowers retailer ability offer easy “i...   
3   helping global network developers journalists ...   
4   biobot analytics analyzes sewage estimate opio...   
5   april netclerk inc acquired builders informati...   
6   satellier one world leading authorities visual...   
7   medcases focuses development medical e learnin...   
8   speeddate com one slew new online webcam enhan...   
9   raven biotechnologies inc biotechnology compan...   
11  xtremespectrum developer ultrawideband uwb imp...   
12  windmil therapeutics developing cell therapies...   
13  alacra organizes financial legal information a...   
14  petdesk saas appointment reminder platform hel...   
15  independent unbiased ally helping employees re...   
16  comstellar technologies inc building major tec...   
17  membership invitation website wine epicurean p...   
19  homebase helps reduce administrative burden ho...   
20  miro leading visual collaboration platform cro...   
21  tenebril inc leading security privacy technolo...   

                                     tokenized_word_1  probab_1  \
0   [codementor, provides, live, help, software, d...  0.035720   
1   [agshift, solution, blends, deep, learning, co...  0.044779   
2   [shipsi, empowers, retailer, ability, offer, e...  0.032292   
3   [helping, global, network, developers, journal...  0.074074   
4   [biobot, analytics, analyzes, sewage, estimate...  0.073395   
5   [april, netclerk, inc, acquired, builders, inf...  0.075657   
6   [satellier, one, world, leading, authorities, ...  0.058811   
7   [medcases, focuses, development, medical, e, l...  0.108108   
8   [speeddate, com, one, slew, new, online, webca...  0.061856   
9   [raven, biotechnologies, inc, biotechnology, c...  0.116279   
11  [xtremespectrum, developer, ultrawideband, uwb...  0.318374   
12  [windmil, therapeutics, developing, cell, ther...  0.142857   
13  [alacra, organizes, financial, legal, informat...  0.070202   
14  [petdesk, saas, appointment, reminder, platfor...  0.071430   
15  [independent, unbiased, ally, helping, employe...  0.093750   
16  [comstellar, technologies, inc, building, majo...  0.038461   
17  [membership, invitation, website, wine, epicur...  0.000000   
19  [homebase, helps, reduce, administrative, burd...  0.000000   
20  [miro, leading, visual, collaboration, platfor...  0.031037   
21  [tenebril, inc, leading, security, privacy, te...  0.049180   

                                        tag_1  
0                          Biopharmaceuticals  
1                                    Software  
2                Wholesale trade and shipping  
3                           Media and content  
4   Medical software and information services  
5                              Personal goods  
6                                    Software  
7                         Healthcare services  
8               Consumer information services  
9                          Biopharmaceuticals  
11                             Semiconductors  
12                         Biopharmaceuticals  
13                                   Software  
14                                   Software  
15        Financial institutions and 

## Conclusion after comparision :

1. For each matching rows from both dataframe, I have visited the site to cross check Industry_segment and find that company description (long)(company_long) works well compare to short_decription column (company_short).
2. Both dataframe have different number of rows.

In [35]:
# converting df to .csv
# downloading into local machine

company_long.to_csv('long_descrip.csv')
#files.download('long_descrip.csv')

In [36]:
company_short.to_csv('short_descrip.csv')
#files.download('short_descrip.csv')

## References :
1. Stackoverflow.
2. Geeks for Geeks.